homeassistant/components/wled/light.py

```diff
@@ -259,3 +275,46 @@ async def async_effect(
             data[ATTR_SPEED] = speed
 
         await self.coordinator.wled.light(**data)
+
+
+@callback
+def async_update_segments(
+    entry: ConfigEntry,
+    coordinator: WLEDDataUpdateCoordinator,
+    current: Dict[int, WLEDLight],
+    async_add_entities,
+) -> None:
+    """Update segments."""
+    segment_ids = {light.segment_id for light in coordinator.data.state.segments}
+    current_ids = set(current)
+
+    # Process new segments, add them to Home Assistant
+    new_segments = []
+    for segment_id in segment_ids - current_ids:
+        current[segment_id] = WLEDLight(entry.entry_id, coordinator, segment_id)
+        new_segments.append(current[segment_id])
+
+    if new_segments:
+        async_add_entities(new_segments)
+
+    # Process deleted segments, remove them from Home Assistant
+    asyncio.gather(
```

Comment:
I don't think this works since we can't await `asyncio.gather` here. Sorry about the extra work.

Reply:
It was a separate commit, so easily reverted.